### PR TITLE
Cleaned up personal settings page

### PR DESCRIPTION
**Pull request by andreasjacobsen93**
I decided to freshen up the personal settings page a bit.

capture
The old vs the new on the right

### CODE REVIEW

**Comment by schiessle** [comment]
cc @nextcloud/designers

**Comment by skjnldsv** [comment]
I like it a lot! :)

**Comment by enoch85** [comment]
Nice! :+1:

File: settings/templates/personal.php [hunk 1]
```diff
@@ -44,7 +44,7 @@
 	<form id="avatarform" class="section" method="post" action="<?php p(\OC::$server->getURLGenerator()->linkToRoute('core.avatar.postAvatar')); ?>">
 		<h2>
 			<label><?php p($l->t('Profile picture')); ?></label>
-			<span class="icon-password"/>
+				<span class="icon-avartar-federation-menu icon-password">&#9662;<span/>
```
Comment (nickvergessen) [inline settings/templates/personal.php]: typo in avatar?

File: settings/css/settings.css [hunk 2]
```diff
@@ -2,20 +2,47 @@
...
+
+.icon-federation-menu {
+	width: 44px;
+	margin: -6px;
+	float: right;
+}
+
+.icon-avartar-federation-menu {
+	position: absolute;
+    width: 44px;
```
Comment (nickvergessen) [inline settings/css/settings.css]: spaces and tabs mixed?

**Comment by andreasjacobsen93** [comment]
@nickvergessen updated, thanks :)

**Comment by MariusBluem** [comment]
27603548-b3c1519c-5b76-11e7-9d0c-02370657c938

**Comment by enoch85** [comment]
@MariusBluem You last point about unstructured I don't agree with. But it would be great if we could do something about that whitespace.

**Comment by Dennis1993** [comment]
See 4688 it's the same idea

**Comment by andreasjacobsen93** [comment]
Update
capture

**Comment by jancborchardt** [comment]
Cool detail fixes! Some feedback:

- This visually mixes personal profile info with personal _settings_, two different things. So putting groups, language and password in columns/boxes as well is good. However I would split that from the profile info and have some vertical whitespace between.
- The triangle is good to show that it's a menu. Can you put it to the right of the icon though because that's the way we (and others) do it elsewhere?
- While aligning all the icons to the right looks good regarding balance, I have a small concern that the move from the heading text disassociates it from the actual element. But let's try it. :)
- Another small fix would be to reduce the whitespace between the headers of Full name, email etc and their respective input fielda just a bit (maybe just by a third, or even half it) - they don't seem very connected at the moment.

**Comment by andreasjacobsen93** [comment]
Thank for the feedback, @jancborchardt 👍 
capture

**Comment by enoch85** [comment]
What about this whitespace, could we use it some how? 
27683530-8915f8f2-5cc6-11e7-9263-a45cbcfa9f1b

**Comment by jancborchardt** [comment]
@andreasjacobsen93 just one more thing: the password-language-groups block should start in the left column. It has nothing to do with the user image so shouldn't be aligned right of it. :)

**Comment by MorrisJobke** [comment]
@andreasjacobsen93 Looks really good! Once the latest remark from @jancborchardt is fixed we can get this in 🎉

**Comment by andreasjacobsen93** [comment]
Update

capture

**Comment by jancborchardt** [comment]
Almost :) now put language and password in the middle and right column so they are on the same row as password. Additionally add some vertical whitespace between the personal profile info and that.

**Comment by schiessle** [comment]
@andreasjacobsen93 great work! 😃 

But you should rebase to the latest master. settings/template/personal.php is no longer in use. Instead you need to apply your changes on the template to settings/templates/settings/personal/personal.info.php

**Comment by andreasjacobsen93** [comment]
cc @jancborchardt 
capture

@schiessle how do I do that? Do I need to make a new pull request?

**Comment by pixelipo** [comment]
Whitespace under the profile picture (mentioned by @enoch85 ) gives better esthetic's while still displaying equal amount on information on a single screen, as seen in the original comment:
capture

**Comment by schiessle** [comment]
@andreasjacobsen93 

````
git checkout master
git pull
git checkout <your branch>
git rebase master
````

then undo your changes to settings/templates/personal.php and apply it to settings/templates/settings/personal/personal.info.php. Should be straight forward the content didn't changed. It was just split up, so you should be able to copy & paste your changes.

If you don't feel familiar enough to git you can also do another pr and close this one.

**Comment by jancborchardt** [comment]
Looks all good to me now! :) Great work @andreasjacobsen93!

**Comment by schiessle** [comment]
I found one small issue. If you go to the admin "sharing" settings and disable "Allow users to publish their data to a global and public address book" the scope selector should be completely gone but in your version the triangle is still there:

personal

**Comment by andreasjacobsen93** [comment]
Update 😄 

capture2

**Comment by skjnldsv** [comment]
Very nice!!

**Comment by MorrisJobke** [comment]
The german text is a bit longer and therefore the help text below the email input is too long causing the box to wrap to the next line:

bildschirmfoto 2017-07-24 um 12 44 30

Adding this style to the `<em>` element

```css 
width: 17em;
display: inline-block;
```

only makes it a bit better:

bildschirmfoto 2017-07-24 um 12 44 41

@andreasjacobsen93 Could you have a final look into this issue? That you we nice. Beside that it looks really awesome 👍

**Comment by andreasjacobsen93** [comment]
Any ideas on this?

**Comment by pixelipo** [comment]
I can take a look at it and make a commit tomorrow, if I manage to fix an issue with my dev-setup

**Comment by MorrisJobke** [comment]
> edit: didn't manage to fix dev-instance - got this error:

Looks like your apache is not properly set up. Does it have the AllowOverride set properly? This is needed to be set in the real Apache config and not in the .htaccess: See https://docs.nextcloud.com/server/12/admin_manual/installation/source_installation.html#apache-web-server-configuration

**Comment by pixelipo** [comment]
HTML structure was quite messy - and still is. I cleaned it up a bit so that it makes more sense, fixing the label issue in the process. I might have missed some `hidden` elements - please review and check if everything is still in place.

**Comment by MorrisJobke** [comment]
Let me try to rebase this and fix the conflict

**Comment by MorrisJobke** [comment]
> Let me try to rebase this and fix the conflict

Okay - I will not do a rebase - this is a can of worms. Let's keep it simple and do a merge commit. 🙈

**Comment by pixelipo** [comment]
I agree. I noticed a few more bugs to fix - federation-menu mostly. 
Question for @nextcloud/designers - what should be alignment of elements in the "profile picture" section? Left or center? Right now it's a mix of both.

**Comment by MorrisJobke** [comment]
> Question for @nextcloud/designers - what should be alignment of elements in the "profile picture" section? Left or center? Right now it's a mix of both.

What do you mean? I don't understand the question. ./

**Comment by pixelipo** [comment]
@MorrisJobke this is what I'm talking about:
screenshot from 2017-08-04 08-25-55
 action icons are centered, everything else is aligned to the left. Is that correct behavior? Should profile picture be centered?

**Comment by MorrisJobke** [comment]
> action icons are centered, everything else is aligned to the left. Is that correct behavior? Should profile picture be centered?

Center it over the buttons and then let's get this in.

File: settings/js/federationsettingsview.js [hunk 3]
```diff
@@ -192,7 +192,7 @@
 					$icon.removeClass('hidden');
 					break;
 				case 'contacts':
-					$icon.addClass('icon-contacts');
+					$icon.addClass('icon-contacts-dark');
```
Comment (pixelipo) [inline settings/js/federationsettingsview.js]: Ooops, sorry about that :blush:

**Review by pixelipo — APPROVED** [review]
looks/works as expected

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
🚀

**Pull request merged**